### PR TITLE
Show InputPanel on focus and pointer press

### LINKED ISSUE #3639: Should Input Pane display policy be Automatic
from #1919 

Code TODO to Investigate if the CoreTextInputPaneDisplayPolicy should be Manual or Automatic.

Current setting is Manual, however, it's TBD what impact this has with the software keyboard. 
`_editContext.InputPaneDisplayPolicy(Core::CoreTextInputPaneDisplayPolicy::Manual);`

See for more info about the properties: [CoreTextInputPaneDisplayPolicy Enum](https://docs.microsoft.com/en-us/uwp/api/Windows.UI.Text.Core.CoreTextInputPaneDisplayPolicy#fields)

### PR DESCRIPTION
**Pull request by leonMSFT**
## Summary of the Pull Request
I originally thought that setting `TSFInputControl::_editContext.InputPaneDisplayPolicy` to be Automatic would allow the InputPanel to show and hide automatically when `TSFInputControl` gains and loses focus. It doesn't seem to behave that way, so we'll show the InputPanel manually. 

I'll show the panel during `PointerPressedHandler` and during `GotFocusHandler`. A user will have the on-screen keyboard pop up when getting focus, but if they close the keyboard, they can simply re-tap on the terminal to bring it back up.

## PR Checklist
* [x] Closes #3639
* [x] CLA signed.
* [x] Tests added/passed

## Validation Steps Performed
Played on my surfaces book with the on screen keyboard by closing/tapping on the terminal and using the search box.

### CODE REVIEW

File: src/cascadia/TerminalControl/TermControl.cpp [hunk 1]
```diff
@@ -824,6 +825,8 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
         const auto ptr = args.Pointer();
         const auto point = args.GetCurrentPoint(*this);
 
+        InputPane::GetForCurrentView().TryShow();
```
Comment (zadjii-msft) [inline src/cascadia/TerminalControl/TermControl.cpp]: Does `_GotFocusHandler` not get fired automatically after this one? I'd think you'd only need it in the GotFocus one.

Comment (leonMSFT) [reply src/cascadia/TerminalControl/TermControl.cpp]: Nope, it doesn't get fired if we already have focus. This call is here for the case where terminal is in focus, but the user closed the keyboard.

**Comment by leonMSFT** [comment]
Yup, the on screen keyboard only shows up if you _don't_ have a keyboard attached.

File: src/cascadia/TerminalControl/TermControl.cpp (hunk 1 above)
Comment (zadjii-msft) [reply src/cascadia/TerminalControl/TermControl.cpp]: Okay good enough for me. (I'd appreciate adding this as a comment in code, but you do you)

**Comment by leonMSFT** [comment]
@msftbot merge this in 1 minute

**Pull request merged**